File: backend/src/main/java/com/example/springbootfullstackapp/service/DepartmentService.java
package com.example.springbootfullstackapp.service;

import java.util.List;

import com.example.springbootfullstackapp.model.Department;

//DepartmentService.java

public interface DepartmentService {

 List<Department> getAllDepartments();

 Department getDepartmentById(Long id);

 Department createDepartment(Department department);

 Department updateDepartment(Long id, Department department);

 void deleteDepartment(Long id);
}

File: frontend/src/services/DepartmentService.js
import axios from 'axios';

const BASE_URL = "http://localhost:8080/api/v1/departments";

class DepartmentService {

    static instance = new DepartmentService();
    static getInstance() {
        return this.instance;
    }
    getAllDepartments(){
        return axios.get(BASE_URL);
    }
    
    createDepartment(department){
        return axios.post(BASE_URL, department);
    }

    getDepartmentById(depId){
        return axios.get(BASE_URL + '/' + depId);
    }
    updateDepartment(department, departmentId){
        return axios.put(BASE_URL + '/' + departmentId, department);
    }

    deleteDepartment(departmentId){
        return axios.delete(BASE_URL + '/' + departmentId);
    }
}

export default DepartmentService.getInstance();